writer_test.go

```diff
@@ -151,6 +164,37 @@ func testWriterMaxAttemptsErr(t *testing.T) {
 	}
 }
 
+func testWriterMaxBytes(t *testing.T) {
```

Comment:
Let's use a context with a timeout instead of `context.Background`.  That will defend against cases where the `WriteMessages` call may hang.

Also, it would be great if you could `createTopic` instead of using constant topic names to ensure that we always start from a clean slate.  That's more of import for folks testing locally than for CI.

Reply:
Great ideas!